File: dotnet-desktop-guide/net/winforms/controls-design/snippets/extend-existing/vb/CustomControl1.vb
'<control>
Public Class CustomControl1

    '<counter>
    Private _counter As Integer = 0
    '</counter>

    '<onpaint>
    Protected Overrides Sub OnPaint(ByVal e As System.Windows.Forms.PaintEventArgs)

        ' Draw the control
        MyBase.OnPaint(e)

        ' Paint our string on top of it
        e.Graphics.DrawString($"Clicked {_counter} times", Font, Brushes.Purple, New PointF(3, 3))

    End Sub
    '</onpaint>

    '<onclick>
    Protected Overrides Sub OnClick(e As EventArgs)

        ' Increase the counter and redraw the control
        _counter += 1
        Invalidate()

        ' Call the base method to invoke the Click event
        MyBase.OnClick(e)

    End Sub
    '</onclick>

End Class
'</control>

File: dotnet-desktop-guide/winforms/controls-design/snippets/extend-existing/csharp/CustomControl1.cs
namespace CustomControlProject;

//<control>
public partial class CustomControl1 : Button
{
    //<counter>
    private int _counter = 0;
    //</counter>

    public CustomControl1()
    {
        InitializeComponent();
    }

    //<onpaint>
    protected override void OnPaint(PaintEventArgs pe)
    {
        // Draw the control
        base.OnPaint(pe);

        // Paint our string on top of it
        pe.Graphics.DrawString($"Clicked {_counter} times", Font, Brushes.Purple, new PointF(3, 3));
    }
    //</onpaint>

    //<onclick>
    protected override void OnClick(EventArgs e)
    {
        // Increase the counter and redraw the control
        _counter++;
        Invalidate();

        // Call the base method to invoke the Click event
        base.OnClick(e);
    }
    //</onclick>
}
//</control>
